test/user.test.js

```diff
@@ -363,6 +363,88 @@ describe('User', function() {
     });
   });
 
+  describe('Password length validation', function() {
+    var pass72Char = new Array(73).join('a');
+    var pass73Char = new Array(74).join('a');
+    var passTooLong = pass72Char + 'WXYZ1234';
+
+    it('rejects passwords longer than 72 characters', function(done) {
+      try {
+        User.create({ email: 'b@c.com', password: pass73Char }, function(err) {
+          if (err) return done (err);
+          expect(err).to.have.property('message', /Password too long/);
+        });
+      } catch (e) {
+        done();
+      }
```

Comment:
Something is still weird here. If we expect error in the callback API, why do we need try catch. We talked about this over hangout last time @loay. IMO, we should be try/catching before we return the response to the user (they should only have to handle the `err` returned in the callback. If you decide the try/catch is valid, then we should at least have an assertion in 378 to ensure when the error is throw that we assert it's the right type of error being caught -- `assert(e, undefined, 'some-fail-message', '=')`.

@bajtos Thoughts?